kms/src/main/java/io/kroxylicious/kms/provider/kroxylicious/inmemory/InMemoryEdekSerde.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.kroxylicious.inmemory;
+
+import java.nio.ByteBuffer;
+
+import io.kroxylicious.kms.service.De;
+import io.kroxylicious.kms.service.Ser;
+
+class InMemoryEdekSerde implements Ser<InMemoryEdek>, De<InMemoryEdek> {
+
+    @Override
+    public InMemoryEdek deserialize(ByteBuffer buffer) {
+        var numAuthBits = buffer.getShort();
+        var ivLength = buffer.getShort();
```

Comment:
ivLengths for AES-GCM about 96bit, so a single byte would still be more than adequate.

Reply:
In practice, yes, 96 bits is recommended. The question is whether we should assume/enforce that _here_. I don't think we need to, because this is just the in-memory (i.e. testing) KMS. We could use a Ceasar cipher and it would be fine. In fact it's kind-of confusing that this is using AES at all. 

Relatedly, perhaps we should have a separate module for this KMS precisely so that it ends up a bit harder for people to configure the in-memory KMS by accident?